private/model/api/docstring.go

```diff
@@ -88,21 +92,134 @@ func docstring(doc string) string {
 	doc = reComments.ReplaceAllString(doc, "")
 	doc = reFullname.ReplaceAllString(doc, "")
 	doc = reExamples.ReplaceAllString(doc, "")
-	doc = rePara.ReplaceAllString(doc, "$1\n\n")
-	doc = reLink.ReplaceAllString(doc, "$2 ($1)")
-	doc = reTag.ReplaceAllString(doc, "$1")
+
+	_, err := xhtml.Parse(strings.NewReader(doc))
+	// If there is no error, means it is HTML. However, if it isn't HTML, we use the old
+	// way of generating documentation.
+	if err == nil {
+		// this will only occur if an error during the tokenization process occurs.
+		doc = generateDocFromHTML(doc)
+	} else {
+		doc = rePara.ReplaceAllString(doc, "$1\n\n")
+		doc = reLink.ReplaceAllString(doc, "$2 ($1)")
+	}
+
 	doc = reEndNL.ReplaceAllString(doc, "")
-	doc = strings.TrimSpace(doc)
 	if doc == "" {
 		return "\n"
 	}
 
 	doc = html.UnescapeString(doc)
-	doc = wrap(doc, 72)
-
 	return commentify(doc)
 }
 
+// generateDocFromHTML will generate the proper doc string for html encoded doc entries.
+func generateDocFromHTML(htmlSrc string) string {
+	tokenizer := xhtml.NewTokenizer(strings.NewReader(htmlSrc))
+	// tagInfo contains necessary token info of start tag
+	type tagInfo struct {
+		tag string
+		key string
+		val string
+		txt string
+	}
+	doc := ""
+	level := 0
+	col := 0
+	indent := false
+	stack := []tagInfo{}
+
+	for tt := tokenizer.Next(); tt != xhtml.ErrorToken; tt = tokenizer.Next() {
+		switch tt {
+		case xhtml.ErrorToken:
+			break
+		case xhtml.TextToken:
+			txt := string(tokenizer.Text())
+			if len(stack) > 0 {
+				// set the current entries txt value. We use this
+				// for checking if we need to indent. The indent rules
+				// is if there is nested html tags that aren't empty or space,
+				// we will indent the code or li blocks.
+				size := len(stack)
+				stack[size-1].txt = txt
```

Comment:
Yes, already changed. Was because the stack came later when I realized I needed one